pkg/nfd-master/nfd-api-controller.go

```diff
@@ -47,13 +48,15 @@ type nfdController struct {
 type nfdApiControllerOptions struct {
 	DisableNodeFeature bool
 	ResyncPeriod       time.Duration
+	AllowedNamespaces  []string
```

Comment:
Hmm, looking at this now I think we could do better (more k8s-like). Could we use [metav1.LabelSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#labelselector-v1-meta) directly. It would make it very flexible to pick up the namespaces to watch. The name of the config option would be something like `nodeFeatureNamespaceSelector`.  WDYT?

Reply:
If I got this correctly, users would need to pass an array of maps to specify selectors, right?

Reply:
Or even use the metav1.LabelSelector type directly?